pgcli/pgexecute.py

```diff
@@ -11,6 +12,9 @@
 psycopg2.extensions.register_type(psycopg2.extensions.UNICODE)
 psycopg2.extensions.register_type(psycopg2.extensions.UNICODEARRAY)
 
+# When running a query, make pressing CTRL+C raise a KeyboardInterrupt
+psycopg2.extensions.set_wait_callback(psycopg2.extras.wait_select)
```

Comment:
This was explained here: http://initd.org/psycopg/articles/2014/07/20/cancelling-postgresql-statements-python/

Note that wait_select is supposed to carry out the same role as the function defined in that article.